src/ExecutorC/StatementExecutor.c

```diff
@@ -140,7 +141,7 @@ static void destroyMethods(MethodNode* head) {
 }
 
 char* StatementExecutor_Make(StatementExecutor* executor, char const* instanceName, char const* className, SlimList* args){
-	FixtureNode* fixtureNode = findFixture(executor, className);
+	FixtureNode* fixtureNode = replaceSymbolsAndFindFixture(executor, className);
```

Comment:
I don't love the "and" in the function name.  Maybe the new function could be called "findFixture" and the old function could be renamed something like "lookupFixture."  The "and" implies two responsibilities.  The "and" is kind of a single responsibility principle smell.   Thoughts?

Reply:
Hi @dougbradbury. I have found this project very useful in system testing a substantial C and C++ code base with FitNesse, so I am happy to contribute. I really appreciate the time you've taken to review my changes.

I am not very pleased with my function names either. I was trying to find a way to make it clear that className may contain symbols. What about changing
```c
FixtureNode * findFixture(StatementExecutor* executor, char const * className);
```
to
```c
FixtureNode * findFixtureByName(StatementExecutor* executor, char const * className);
```
and
```c
FixtureNode* replaceSymbolsAndFindFixture(StatementExecutor* executor, const char* className);
```
to
```c
FixtureNode* findFixture(StatementExecutor* executor, const char* classNameWithSymbols);
```
?
I am happy to take your suggestion as is, if you prefer, but I was thinking that it might be tricky to distinguish between "findFixture" and "lookupFixture" based on the function names. I'm not sure "findFixture" and "findFixtureByName" are much better though.

Let me know which way you prefer to go or if you have a different suggestion, and I will make the changes. It shouldn't take more than a few minutes. Thanks again!

Reply:
I like your new names much better than mine!